File: Practica_1/src/gestorAplicacion/usuarios/Sueldo.java
package gestorAplicacion.usuarios;

import java.util.*;

/*
* Se agrupan metodos que utiliza la clase empleado para heredar a sus hijas
* Define algunas constantes aplicables para todos los empleados
* cada Hijo de empleado define de forma distinta los metodos aqui declarados
* */

public interface Sueldo {
    public final static double sueldoBase=100;
    public final static double porcentajeComisionProductos=0.2;
    public final static double porcentajeComisionServicios=0.1;
    public abstract double getSueldo();
    public abstract void setSueldo(double sueldo);
    public abstract void aumentarSueldo(double porcentaje);
    public abstract double calculoDePrima();



    


}

File: Python/gestorAplicacion_py/usuarios/Sueldo.py
from abc import ABC, abstractclassmethod

class Sueldo(ABC):
    sueldoBase=100
    porcentajeComisionProductos=0.2
    porcentajeComisionServicios=0.1
    
    def getSueldo(self):
        return self.sueldoBase
    def setSueldo(self,sueldo):
        self.sueldoBase = sueldo

    @abstractclassmethod
    def aumentarSueldo(self):
        pass
    @abstractclassmethod
    def calcularPrima(self):
        pass

# miren si lo siguiente es necesario.
class aumento(Sueldo):
    def __init__(self):
        self.sueldoBase = Sueldo.sueldoBase

    def aumentarSueldo(self):
        aumento = self.sueldoBase * 0.2 + self.sueldoBase *0.1
        return self.sueldoBase + aumento